packages/template-retail-react-app/app/hooks/use-product-view-modal.js

```diff
@@ -5,30 +5,54 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {useEffect, useState} from 'react'
+import {useEffect} from 'react'
 import {rebuildPathWithParams, removeQueryParamsFromPath} from '../utils/url'
 import {useHistory, useLocation} from 'react-router-dom'
-import {useCommerceAPI} from '../commerce-api/contexts'
-import {isError} from '../commerce-api/utils'
 import {useVariant} from './use-variant'
 import {useToast} from './use-toast'
 import {useIntl} from 'react-intl'
 import {API_ERROR_MESSAGE} from '../constants'
+import {useProduct} from 'commerce-sdk-react-preview'
 
 /**
- * This hooks is responsible for fetching a product detail based on the variation selection
+ * This hook is responsible for fetching a product detail based on the variation selection
  * and managing the variation params on the url when the modal is open/close
  * @param initialProduct - the initial product when the modal is first open
  * @returns object
  */
 export const useProductViewModal = (initialProduct) => {
     const location = useLocation()
-    const api = useCommerceAPI()
     const history = useHistory()
     const intl = useIntl()
-    const [product, setProduct] = useState(initialProduct)
-    const [isFetching, setIsFetching] = useState(false)
     const toast = useToast()
+    const urlParams = new URLSearchParams(location.search)
+    const pid = urlParams.get('pid')
+    // use commerce-sdk-react-preview useProduct to fetch new product detail whenever variant changes
```

Comment:
I don't think this needs a comment, the code is pretty self-documenting here

Reply:
Actually... does this re-run when the variant changes? We're not in a `useEffect` here so there is an open question about whether a query param change triggers an entire page re-render. Without data binding to `?pid=<id>` via 

```javascript
useEffect( () => {
  // respond to pid changes
}, [pid])
``` 

How will the state update?

Reply:
It does re-run because we pass `pid` to `id` for `useProduct`, which is treated as queryKey. React query is smart enough to re-fetch the data when queryKey changes. However, there is a downside to this approach is that there will be some extra renders because pid can only change after variant changes (this update the url with correct pid)
To be precise, when the modal is first mounted (user clicks on Edit) and click on swatches options

- Swatch options will update url with correct size/color => the url still has old pid
- This will trigger variant to change => re-update url with correct `pid`
- When the pid changes, react-query will trigger re-fetch to update the latest data

The reason we need to refetch product detail with new pid is to be able to get the stock level on each variant
All of this happens in the background since there is no indicator on the UI.

Reply:
I am testing an approach that uses internal state to handle this flow. It could be more performant imo 🤔